src/BrandEmbassyCodingStandard/Sniffs/Classes/ForbiddenStaticMethodSniff.php

```diff
@@ -0,0 +1,174 @@
+<?php declare(strict_types = 1);
+
+namespace BrandEmbassyCodingStandard\Sniffs\Classes;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use ReflectionClass;
+use ReflectionMethod;
+use SlevomatCodingStandard\Helpers\ClassHelper;
+use SlevomatCodingStandard\Helpers\StringHelper;
+use SlevomatCodingStandard\Helpers\TokenHelper;
+use function array_map;
+use function in_array;
+use function is_a;
+use function sprintf;
+use const T_CLASS;
+use const T_DOUBLE_COLON;
+use const T_SELF;
+use const T_STATIC;
+use const T_STRING;
+
+final class ForbiddenStaticMethodSniff implements Sniff
+{
+    private const FORBIDDEN_SELF_METHOD_CALL = 'ForbiddenSelfMethodCall';
+
+    /**
+     * @var array<string, class-string>
+     */
+    public $forbiddenClasses = [];
+
+
+    /**
+     * @return int[]|string[]
+     */
+    public function register(): array
+    {
+        return [
+            T_SELF,
+            T_STATIC,
+        ];
+    }
+
+
+    /**
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint
+     *
+     * @param int $stackPtr
+     */
+    public function process(File $phpcsFile, $stackPtr): void
+    {
+        $classPointer = TokenHelper::findPrevious($phpcsFile, [T_CLASS], $stackPtr);
```

Comment:
Není lepší použít `\SlevomatCodingStandard\Helpers\ClassHelper::getClassPointer`?

Tvůj přístup bude blbnout když někde nad tím použiješ anonymní třídu (což třeba já občas v testech dělávám).

Reply:
Mel sem to tak, ale z nejakeho duvodu to nefunguje tak jak bych si predstavoval...asi tam dela binec ten scope ve kterem to hleda... nevim

Reply:
OK, je to asi zbytečný edge-case... Takhle to imho stačí - při nejhorším to nechá `self::asert` někde, kde ho nechceme, pokud nad tím bude definovaná jiná anonymní třída.